DSA Transparency Extension

lamrowena opened the pull request:
Specification for DSA Transparency submitted by the Global Privacy Working Group

dennyd committed:
added anonymised custom audiences

hillslatt committed:
segtax id reservation for anonymised.io

Added anonymised custom audiences

hillslatt committed:
Merge pull request #146 from InteractiveAdvertisingBureau/develop

Chrome Cookie Deprecation Ext

hillslatt committed:
Merge pull request #148 from InteractiveAdvertisingBureau/develop

Chrome Testing Ext update

lamrowena committed:
Create dsa_transparency.md

Extension for DSA Transparency

lamrowena committed:
Formatting

Updated formatting

lamrowena committed:
Fix hyperlinks

lamrowena committed:
Typo in bidrequest

Corrected typo in the bid request sample. "dsatopub" updated to  "datatopub"

sbelov commented on extensions/community_extensions/dsa_transparency.md:
```diff
@@ -0,0 +1,345 @@
+<tr>
+<td><code>transparency</code></td>
+<td>array of object</td>
+<td>Array of objects of the entities that applied user parameters and the parameters they applied.</td>
```
Could these entities and parameters they applied differ on a per-impression (ad slot) basis in terms of the sellside targeting? 
If so, should this transparency information be part of the `Imp` object?

lamrowena replied:
The GPWG discussed and would like to tackle this post-MVP which this extension is meant to be.

MRuecker replied:
Please start values with 1 and leave 0 for undknown/not set

sbelov commented on extensions/community_extensions/dsa_transparency.md:
```diff
@@ -0,0 +1,345 @@
+<tr>
+<td><code>pubrender</code></td>
+<td>integer</td>
+<td>Flag to indicate if the publisher will render the DSA Transparency info. This will signal if the publisher is able to and intends to render the icon and display the DSA transparency info to the end user.
```
Could a publisher's ability to render depend on a specific `Imp` object, in case multiple impressions are offered through a single bid request? If that's the case, should this flag reside somewhere under `Imp` object?

For example… a publisher might be able to render the transparency information for native ads, but not for HTML banners.

lamrowena replied:
Similar to the comment above, the GPWG discussed and would like to tackle this post-MVP which this extension is meant to be.

lamrowena committed:
DSA Transparency amended public comment period end date

Amended public comment period end date

SyntaxNode commented on extensions/community_extensions/dsa_transparency.md:
```diff
@@ -0,0 +1,345 @@
+    "regs": {
+   	 "ext": [{
+   		 "dsa": [{
+            "dsainfo": 3, 
```
Please fix the indentation of the `dsa` object for ease of reading.

lamrowena replied:
Done!

SyntaxNode commented on extensions/community_extensions/dsa_transparency.md:
```diff
@@ -0,0 +1,345 @@
+    "regs": {
+   	 "ext": [{
+   		 "dsa": [{
+            "dsainfo": 3, 
```
Do we need to repeat the `dsa` name in `dsainfo`? Can it simply be `info`?

lamrowena replied:
The GPWG reviewed and has since updated this to `required` since info could be misleading as this indicates whether the dsa information is needed and not the actual dsa information itself.

sergseven replied:
`required` is considered as a unfortunate naming as well, since keyword `required` is a reserved one in many programming languages as well as in protobuf v2 and thrift:

proto v2 https://protobuf.dev/reference/protobuf/proto2-spec/#fields
thrift https://thrift.apache.org/docs/idl.html#field-requiredness
Please consider to rename it to something else, otherwise it will be very painful to implement json parsing.

lamrowena committed:
Updated formatting 

Updated formatting of dsa object in sample bid request for ease of reading

zbartholomew commented on extensions/community_extensions/dsa_transparency.md:
```diff
@@ -0,0 +1,344 @@
+    ],
+    "regs": {
+        "ext": [{
+            "dsa": [{
```
Above we say the ext.dsa is an object but here in the example, we show it as an array of objects.  Am I reading the spec correctly?  If so can we update the example to reflect an object not an array of dsa objects?

lamrowena replied:
Corrected

kristijanexads replied:
@lamrowena shouldn't `regs.ext` also be an object? All oRTB docs have `ext` as object, not array of object (applies to both bid request and bid response).

tkanos commented:
Guys, the DSA act is almost there. 

For now we only have https://iabeurope.eu/wp-content/uploads/2023/10/DSA-Transparency-Approach-Summary-Doc.pdf as documentation, and it differs a bit with that one. 

It will be good to have it finalized and merged.

lamrowena committed:
Update dsa_transparency for public comments.md

- Updated dsainfo attribute
- Fixed sample bid request and bid response so that dsa is an object and not object array
- Additional url parameters and macros for parity of information sent in the bid request and response

kristijanexads commented on extensions/community_extensions/dsa_transparency.md:
```diff
@@ -0,0 +1,379 @@
+   		 ],
+   		 "ext": [{
+   			 "dsa": {
+   				 "behalf": "Advertiser",
```
Based on [https://iabeurope.eu/wp-content/uploads/2023/10/DSA-Transparency-Approach-Summary-Doc.pdf](https://iabeurope.eu/wp-content/uploads/2023/10/DSA-Transparency-Approach-Summary-Doc.pdf) this should also go under `transparency` I assume?

DSA summary doc:
```
seatbid.bid.ext.transparency.behalf
seatbid.bid.ext.transparency.paid
seatbid.bid.ext.transparency.userparameters
```
PR:
```
seatbid.bid.ext.dsa.behalf
seatbid.bid.ext.dsa.paid
seatbid.bid.ext.dsa.transparency.params
```
Suggested:
```
seatbid.bid.ext.dsa.transparency.behalf
seatbid.bid.ext.dsa.transparency.paid
seatbid.bid.ext.dsa.transparency.userparameters
```
Beside that I suggest `ext` should be an object, not array of objects since changing `ext` to array of objects wouldn't be in line with official oRTB docs.

lamrowena replied:
Changed `ext` to an object instead of array. The transparency object array is of the domains and the user parameters applied per domain.

MRuecker replied:
Please start values with 1 and leave 0 for unknown/not set

kristijanexads commented on extensions/community_extensions/dsa_transparency.md:
```diff
@@ -0,0 +1,379 @@
+        "USD"
+    ],
+    "regs": {
+        "ext": [{
```
`ext` should be an object, instead array of objects, to be aligned with oRTB docs.

lamrowena replied:
Fixed

kristijanexads commented on extensions/community_extensions/dsa_transparency.md:
```diff
@@ -0,0 +1,379 @@
+                "datatopub": 2,
+                "transparency": [{
+                    "domain": "platform1domain.com",
+                    "params": [1]},
```
Same as in the comment for the response, either `params` or `userparameters`, as long as it's the same for both

lamrowena committed:
Update dsa_transparency.md

Updated so that sample does not show extension being an object array

sergseven commented on extensions/community_extensions/dsa_transparency.md:
```diff
@@ -0,0 +1,379 @@
+<td><strong>Description</strong></td>
+</tr>
+<tr>
+<td><code>required</code></td>
```
[critical]

Word `required` is a reserved one in many programming languages as well as in protobuf v2 and thrift:
* proto v2 https://protobuf.dev/reference/protobuf/proto2-spec/#fields
* thrift https://thrift.apache.org/docs/idl.html#field-requiredness
Please consider to rename it to something else, otherwise it will be very painful to implement json parsing.

```suggestion
<td><code>requireness</code></td>
```

patmmccann replied:
How about req

kristijanexads replied:
Or `mandatory`

simontrasler replied:
+1 please do not use `required`.

bukrain commented on extensions/community_extensions/dsa_transparency.md:
```diff
@@ -0,0 +1,379 @@
+   			 "dsa": {
+   				 "behalf": "Advertiser",
+   				 "paid": "Advertiser",
+   				 "transparency": {
```
I think it would be a good idea to change transparency in the example to array of objects instead of an object, as it's defined in specification as "array of object".

Lynnleung999 commented:
if the bid.ext.dsa.transparency is null, then, how to disclosure main parameters to end-user?

simontrasler commented:
> if the bid.ext.dsa.transparency is null, then, how to disclosure main parameters to end-user? is this legal?

This forum only relates to the signaling mechanism -- the fields are either populated or they are not. :-) For your obligations, please consult your legal counsel.

M-Hassan90s reviewed (commented):
ok

Jbgaming318 reviewed (commented):
made changes

impeccableinfinity reviewed (approved):
Prototype all mu rights

Ivanreyes9 reviewed (changes requested):
Help
